helfi_tunnistamo.module

```diff
@@ -75,4 +75,14 @@ function helfi_tunnistamo_openid_connect_userinfo_alter(
   if (empty($userinfo['email'])) {
     $userinfo['email'] = helfi_tunnistamo_create_email($userinfo);
   }
+
+  // Unset preferred_username if it is a UUID so that new users get their
+  // name from `name` field instead.
+  if (
+    !empty($userinfo['preferred_username'])
+    && preg_match('/^\w{8}-\w{4}-\w{4}-\w{4}-\w{12}$/', $userinfo['preferred_username'])
```

Comment:
The `preg_match` check can be replaced with `Uuid:isValid()`: https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Component%21Uuid%21Uuid.php/function/Uuid%3A%3AisValid/10.0.x